### PR TITLE
Initial ASB Design Doc

### PR DESCRIPTION
**Pull request by eriknelson**
Rough design doc intro to get some of our thoughts down. I expect this to become an iterative doc as we explore the space more.

### CODE REVIEW

File: docs/design.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+provision == `PUT /v2/service_instances/:instance_id`
+
+Because the Service Catalog is the source of truth for service instances and bindings,
+it provides the ASB with an ID for a desired service instance. The ASB is responsible
+for whatever bookkeeping is necessary to make sure it can perform the requested operations
+when given this ID. Likely needs to be some kind of GUID.
+
+Puts a record of the instance in its store with whatever bookkeeping 
+data is required, then tells the relevant AA to `provision` itself with the
+user provided parameters given to the ASB via the provision request. AA is reponsible
```
Comment (jmrodri) [inline docs/design.md]: typo: reponsible => responsible

File: docs/design.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+the Service Catalog in the form of `parameters` during a provision call.
+
+provision == `PUT /v2/service_instances/:instance_id`
+
+Because the Service Catalog is the source of truth for service instances and bindings,
+it provides the ASB with an ID for a desired service instance. The ASB is responsible
+for whatever bookkeeping is necessary to make sure it can perform the requested operations
+when given this ID. Likely needs to be some kind of GUID.
+
+Puts a record of the instance in its store with whatever bookkeeping
```
Comment (jmrodri) [inline docs/design.md]: trailing white space after bookkeeping is insulting my vim editor :)

File: docs/design.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+* Async? Required, Optional, Unsupported
+
+* Parameters - Schema representing configuration params available to user
+
+
+**Install/Catalog**
+
+ASB pulls inventory of spec files from local store, converts to []Service, sends to Service Catalog
+
+Note: Paremeter handling is still a [topic of discussion](https://github.com/openservicebrokerapi/servicebroker/pull/74)
```
Comment (jmrodri) [inline docs/design.md]: typo: Paremeter => Parameter

File: docs/design.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+### Flow
+
+**Pre-broker install**
+
+It's possible to have registries containing ~15k AAs. On ASB's installation,
+`/catalog` will be called by the Service Catalog, and the ASB needs to respond with
+the inventory of known spec files in the form of Service objects (defined by OSB spec).
+
+Shouldn't be done at installation; ASB needs to be bootstraped so spec files
```
Comment (jmrodri) [inline docs/design.md]: typo: bootstraped => bootstrapped

File: docs/design.md (hunk 2 above)
Comment (eriknelson) [inline docs/design.md]: Mine should have caught this with a big red highlight! Need to check my plugins...

**Pull request closed without merging**